### PR TITLE
Refactor(e2e): Use new data fixture for Festival E2E.

### PR DESCRIPTION
**Pull request by xmano**
Resolve the E2E regression ❤️: issue #3496 Part 1

Note this only addresses part of the issue. We have to fix for all other scripts.

### CODE REVIEW

File: apps/festival/festival-e2e/src/fixtures/user.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// ***********************************************************
+// Primary import for all fixtures
+//
+// All fixtures should support the Query Interface
+// ***********************************************************
+import Users from '../../../../../tools/fixtures/users.json'
```
Comment (RemcoSimonides) [inline apps/festival/festival-e2e/src/fixtures/user.ts:6]: Isn't it possible to do just `tools/fixtures/users.json` here?

File: apps/festival/festival-e2e/src/fixtures/user.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  get(query: QueryInferface) : Partial<UserType>[] {
+    const userSet: Partial<UserType>[]  = (query.exist) ? 
+                                        Users : NewUsers;
+
+    if (query.index && query.index !== -1) {
+      return [userSet[query.index]];
+    }
+
+    if (query.key) {
+      const resultsUser = userSet.filter(u =>
```
Comment (RemcoSimonides) [inline apps/festival/festival-e2e/src/fixtures/user.ts:27]: No need to first use `const` here. You can return the value immediately

File: apps/festival/festival-e2e/src/fixtures/user.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    const userSet: Partial<UserType>[]  = (query.exist) ? 
+                                        Users : NewUsers;
+
+    if (query.index && query.index !== -1) {
+      return [userSet[query.index]];
+    }
+
+    if (query.key) {
+      const resultsUser = userSet.filter(u => 
+                          (u[query.key] === query.value));
```
Comment (RemcoSimonides) [inline apps/festival/festival-e2e/src/fixtures/user.ts:28]: You can remove the extra () here too; `u => u[query.key] === query.value)`

File: apps/festival/festival-e2e/src/integration/dashboard/create-screening-2687.spec.ts [hunk 4]
```diff
@@ -13,135 +13,120 @@ import {
...
+const users  =  [ 
+  (userFixture.get({exist: true, key:'email', value: EVENTS[0].by.email})[0]),
+  (userFixture.get({exist: true, key:'email', value: 'vchoukroun@fake.com'})[0])
+];
 
 describe('User create a screening', () => {
   beforeEach(() => {
     clearDataAndPrepareTest('/');
-    tomorrow = new Date(NOW);
+    tomorrow = new Date(new Date().setDate(NOW.getDate() + 1));
```
Comment (RemcoSimonides) [inline apps/festival/festival-e2e/src/integration/dashboard/create-screening-2687.spec.ts:35]: I see we have momentjs in the package.json. Maybe you can use it here to make this line a lot more readable. You can find the docs here; https://momentjs.com/docs/

If you get it to work then the result will look like this
`let tomorrow  = moment().add(1,'days');`
`let twodayslater  = moment().add(2,'days');`

File: apps/festival/festival-e2e/src/support/pages/marketplace/FestivalScreeningPage.ts [hunk 5]
```diff
@@ -1,28 +1,39 @@
...
+   *   Private Event : Invitation request is sent
+   *   Public Event  : Event is added to user's calendar
+   * @param screeningTitle : Title of event
+   */
+  clickRequestInvitation(screeningTitle: string) {
+    cy.get('festival-screening event-screening-item', {timeout: 3000})
+      .contains(screeningTitle)
+      .parent().parent().parent()
+      .find('button[test-id=invitation-request]').click();
+    //cy.wait(3000);
```
Comment (RemcoSimonides) [inline apps/festival/festival-e2e/src/support/pages/marketplace/FestivalScreeningPage.ts:36]: Remove

File: libs/e2e/src/lib/utils/functions.ts [hunk 6]
```diff
@@ -12,11 +12,24 @@ export function clearDataAndPrepareTest(path: string = '/auth') {
...
-export function signIn(user: Partial<User>) {
+export function signIn(user: Partial<User>, fillIdentity: boolean = false) {
   const p1: AuthLoginPage = new AuthLoginPage();
   p1.switchMode();
   p1.fillSignin(user);
   p1.clickSignIn();
+  cy.wait(10000);
+  //Submit Identity on demand
+  if (fillIdentity) {
+    // cy.url().then(afterLoginURL => {
```
Comment (RemcoSimonides) [inline libs/e2e/src/lib/utils/functions.ts:23]: What's happening here? Saving for later?

**Review by RemcoSimonides — APPROVED** [review]

File: apps/festival/festival-e2e/src/integration/dashboard/create-screening-2687.spec.ts (hunk 4 above)
Comment (xmano) [reply apps/festival/festival-e2e/src/integration/dashboard/create-screening-2687.spec.ts:35]: The package is not there yet in DevDependencies. For now I will use this. When we write some more extensive code we might go over to this.

File: apps/festival/festival-e2e/src/fixtures/user.ts (hunk 1 above)
Comment (xmano) [reply apps/festival/festival-e2e/src/fixtures/user.ts:6]: Yes its possible. Corrected.

File: apps/festival/festival-e2e/src/fixtures/user.ts (hunk 2 above)
Comment (xmano) [reply apps/festival/festival-e2e/src/fixtures/user.ts]: Changed

File: apps/festival/festival-e2e/src/fixtures/user.ts (hunk 3 above)
Comment (xmano) [reply apps/festival/festival-e2e/src/fixtures/user.ts]: Done

File: libs/e2e/src/lib/utils/functions.ts (hunk 6 above)
Comment (xmano) [reply libs/e2e/src/lib/utils/functions.ts:23]: Might need this code for more testing. Will remove once the test stabilizes.

**Pull request merged**